File: collectingcoins.py
t = int(input())

for i in range(t):
    
    s = 0
    a, b, c, n = map(int, input().split())
    mx = max(a, max(b, c))
    s = a + b + c + n

    if s % 3 == 0 and s/3 >= mx:
        print("YES")
    else:
        print("NO")

File: collectingcoins.java
import java.util.Scanner;

public class collectingcoins {
	
	public static void main(String[] args) {

		Scanner in = new Scanner(System.in);

		int t = in.nextInt();

		while(t--> 0) {
			
			int s = 0;
			int a = in.nextInt(), b = in.nextInt(), c = in.nextInt(), n = in.nextInt();
			int max = Math.max(a, Math.max(b, c));
			s = a + b + c + n;

			if(s % 3 == 0 && s / 3 >= max) {

				System.out.println("YES");
			
			}

			else{
			
				System.out.println("NO");

			}
		}
	}
}
